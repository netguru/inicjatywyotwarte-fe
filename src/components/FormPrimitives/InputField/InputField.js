import React from 'react'
import styled from 'styled-components'
import Label from 'components/FormPrimitives/FieldLabel/FieldLabel'
import Error from 'components/FormPrimitives/FieldError/FieldError'

const InputBox = styled.div`
  display: flex;
  flex-direction: ${props => (props.checkboxes ? 'row' : 'column')};
  margin-bottom: 1.5rem;
  width: 100%;
`
InputBox.displayName = "InputBox";

const Input = styled.input`
  height: 3rem;
  font-size: 0.9rem;
  padding: 0 0.8rem;
  font-weight: 100;
  background: ${props => props.theme.lightBackgroundColor};
  border: 1px solid ${props =>
    props.invalid
    ? props.theme.warningColor
    : props.theme.modalColorVeryLight
  };
  border-radius: ${props => props.theme.borderRadiusSmall};
  outline: none;
  color: ${props => props.theme.modalColorLight};

  &:focus,
  &:active {
    border: ${props => props.theme.inputActiveBorder};
  };
`
Input.displayName = "Input";

export default function InputField ({
  name,
  onChange,
  error,
  label,
}) {
  return (
    <InputBox>
      <Label htmlFor={name}>{label}</Label>
      <Input
        id={name}
        invalid={Boolean(error)}
        onInput={() => onChange()}
      />
      {error && <Error className="error">{error}</Error>}
    </InputBox>
  )
}
