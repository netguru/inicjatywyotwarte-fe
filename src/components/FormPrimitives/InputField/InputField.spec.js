import React from 'react'
import InputField from './InputField'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const baseProps = {
  name: 'name',
  onChange: jest.fn(),
  error: null,
  label: 'label'
}

describe('InputField', () => {
  test('renders styled.label', () => {
    const wrapper = shallow(<InputField {...baseProps} />);
    expect(wrapper.find('InputBox').length).toEqual(1);
  });

  test('renders styled.label', () => {
    const propsWithError = {
      ...baseProps,
      error: 'error text message'
    };
    const wrapper = shallow(<InputField {...propsWithError} />);
    expect(wrapper.find('[className="error"]').length).toEqual(1);
    expect(wrapper.find('[className="error"]').text()).toEqual('error text message');
  });


  test('match snapshot', () => {
    const wrapper = shallow(<InputField/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


