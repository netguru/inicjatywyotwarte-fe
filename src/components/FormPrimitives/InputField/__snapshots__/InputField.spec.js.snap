// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField match snapshot 1`] = `
<InputBox>
  <styled.label />
  <Input
    invalid={false}
    onInput={[Function]}
  />
</InputBox>
`;
